selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/theia/TheiaHostedPluginSelectPathForm.java

```diff
@@ -0,0 +1,220 @@
+import static org.eclipse.che.selenium.pageobject.theia.TheiaHostedPluginSelectPathForm.Locators.CLOSE_ICON_XPATH;
+import static org.eclipse.che.selenium.pageobject.theia.TheiaHostedPluginSelectPathForm.Locators.DROP_DOWN_SUGGESTIONS_XPATH;
+import static org.eclipse.che.selenium.pageobject.theia.TheiaHostedPluginSelectPathForm.Locators.DROP_DOWN_SUGGESTION_XPATH_TEMPLATE;
+import static org.eclipse.che.selenium.pageobject.theia.TheiaHostedPluginSelectPathForm.Locators.DROP_DOWN_TITLE_XPATH;
+import static org.eclipse.che.selenium.pageobject.theia.TheiaHostedPluginSelectPathForm.Locators.OPEN_BUTTON_XPATH;
+import static org.eclipse.che.selenium.pageobject.theia.TheiaHostedPluginSelectPathForm.Locators.PROJECT_TREE_ITEM_ID_TEMPLATE;
+import static org.eclipse.che.selenium.pageobject.theia.TheiaHostedPluginSelectPathForm.Locators.ROOT_PROJECTS_FOLDER_XPATH;
+import static org.eclipse.che.selenium.pageobject.theia.TheiaHostedPluginSelectPathForm.Locators.SELECTED_TEM_XPATH_TEMPLATE;
+import static org.eclipse.che.selenium.pageobject.theia.TheiaHostedPluginSelectPathForm.Locators.TITLE_XPATH;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import java.util.ArrayList;
+import java.util.List;
+import org.eclipse.che.selenium.core.webdriver.SeleniumWebDriverHelper;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+@Singleton
+public class TheiaHostedPluginSelectPathForm {
+  private final SeleniumWebDriverHelper seleniumWebDriverHelper;
+
+  @Inject
+  TheiaHostedPluginSelectPathForm(SeleniumWebDriverHelper seleniumWebDriverHelper) {
+    this.seleniumWebDriverHelper = seleniumWebDriverHelper;
+  }
+
+  public interface Locators {
+    String TITLE_XPATH =
+        "//div[@class='dialogBlock']//div[@class='dialogTitle']//div[text()='Hosted Plugin: Select Path']";
+    String CLOSE_ICON_XPATH =
+        "//div[@class='dialogBlock']//div[@class='dialogTitle']//i[contains(@class, 'closeButton')]";
+    String DROP_DOWN_TITLE_XPATH =
+        "//div[@class='dialogBlock']//select[@class='theia-LocationList']";
+    String DROP_DOWN_SUGGESTIONS_XPATH = DROP_DOWN_TITLE_XPATH + "//option";
+    String DROP_DOWN_SUGGESTION_XPATH_TEMPLATE =
+        "(//div[@class='dialogBlock']//select[@class='theia-LocationList']//option)[%s]";
+
+    String ROOT_PROJECTS_FOLDER_XPATH =
+        "//div[@class='dialogBlock']//div[@class='theia-TreeNodeContent']//div[@id='/projects']";
+    String PROJECT_TREE_ITEM_ID_TEMPLATE = "/projects/%s";
+    String SELECTED_TEM_XPATH_TEMPLATE =
+        "//div[@data-node-id='/projects/%s']/parent::div/parent::div[contains(@class, 'theia-mod-selected') and contains(@class, 'theia-mod-focus')]";
+    String CANCEL_BUTTON_XPATH = "//div[@class='dialogBlock']//button[text()='Cancel']";
+    String OPEN_BUTTON_XPATH = "//div[@class='dialogBlock']//button[text()='Open']";
+  }
+
+  private String getSelectedItemXpath(String itemPath) {
```

Comment:
Could you, please, explain what is the difference between _item_ and _suggestion_?